includes/class-sensei-modules.php

```diff
@@ -1543,19 +1543,33 @@ public function admin_enqueue_styles() {
 	 * @return void
 	 */
 	public function course_modules_title() {
-		if ( sensei_module_has_lessons() ) {
-			global $post;
+		if ( ! sensei_module_has_lessons() ) {
+			return;
+		}
 
-			/**
-			 * Filters the module title on the single course page.
-			 *
-			 * @since 2.2.0
-			 *
-			 * @param string $html   The HTML to be displayed.
-			 * @param int $course_id Course ID.
-			 */
-			echo wp_kses_post( apply_filters( 'sensei_modules_title', '<header class="modules-title"><h2>' . __( 'Modules', 'sensei-lms' ) . '</h2></header>', $post->ID ) );
+		/**
+		 * Access check for the course lessons.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param bool $show_lessons Whether or not the lessons should be shown. Default true.
+		 * @param int  $course_id    Course ID.
+		 */
+		if ( ! apply_filters( 'sensei_course_lessons_has_access', true, get_the_ID() ) ) {
```

Comment:
I don't like having to reproduce the filter and its documentation in multiple places. Could we instead create a helper method containing the filter that is called from here?

Reply:
Good call. Fixed in https://github.com/Automattic/sensei/pull/2763/commits/14e2809502dacf0478bdb18585a221f60c7806aa.